### PR TITLE
[REG2.068.0-b2] Issue 14851 - Cannot assign array operation result to static array variable

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=14851

### CODE REVIEW

File: src/expression.c [hunk 1]
```diff
@@ -11572,6 +11572,20 @@ Expression *AssignExp::semantic(Scope *sc)
...
                     return new ErrorExp();
             }
+
+            // e2 matches to t1 because of the implicit length match, so
+            if (isUnaArrayOp(e2x->op) || isBinArrayOp(e2x->op))
+            {
+                // convert e1 to e1[]
+                // e.g. e1[] = a[] + b[];
+                e1x = new SliceExp(e1x->loc, e1x, NULL, NULL);
+                e1x = e1x->semantic(sc);
```
Comment (MartinNowak) [inline src/expression.c]: Why did the but only affect `|` and `^`?

Comment (9rnsr) [inline src/expression.c]: The bug is not only for `|` and `^`. If any array operations appear in e2 of static array assignment, it was broken. For example: `sarr = a[] + b[];` `sarr = a[] * b[];` `sarr = a[] / b[] - v;` ...

Comment (9rnsr) [inline src/expression.c]: Sorry!, I was wrong. Yes, the bug is only for `|`and `^`. The reason is:
The `implicitConvTo()` function is explicitly testing `OrExp` and `XorExp`. If both `e1` and `e2` matches to the target type `t`, the `Or`/`XorExp` also matches to.

Comment (9rnsr) [inline src/expression.c]: But, this fix itself is still correct. In the future, I'm planning to add implicit length check for all array operations. After that, this fix will work more.

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**